3-Authorization-II/protect-web-api/README.md

```diff
@@ -1,226 +1,230 @@
+      * Since this app signs-in users, we will now proceed to select **delegated permissions**, which is requested by apps that sign-in users.
+      * In the **Delegated permissions** section, select **ToDoList.Read**, **ToDoList.ReadWrite** in the list. Use the search box if necessary.
+    1. Select the **Add permissions** button at the bottom.
+
+##### Configure Optional Claims
+
+1. Still on the same app registration, select the **Token configuration** blade to the left.
+1. Select **Add optional claim**:
+    1. Select **optional claim type**, then choose **ID**.
+    1. Select the optional claim **acct**.
+    > Provides user's account status in tenant. If the user is a **member** of the tenant, the value is *0*. If they're a **guest**, the value is *1*.
+    1. Select **Add** to save your changes.
+
+##### Configure the client app (webapp) to use your app registration
 
 Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
 
 > In the steps below, "ClientID" is the same as "Application ID" or "AppId".
 
 1. Open the `webapp\src\main\resources\application.yml` file.
-1. Find the key `Enter_Your_Tenant_ID_Here` and replace the existing value with your Azure AD tenant ID.
-1. Find the key `Enter_Your_Client_ID_Here` and replace the existing value with webApp.AppId.
-1. Find the key `Enter_Your_Client_Secret_Here` and replace the existing value with the key you saved during the creation of `java-spring-webapp-client` copied from the Azure portal.
+1. Find the key `Enter_Your_Tenant_ID_Here` and replace the existing value with your Azure AD tenant/directory ID.
+1. Find the key `Enter_Your_Client_ID_Here` and replace the existing value with the application ID (clientId) of `webapp` app copied from the Azure portal.
+1. Find the key `Enter_Your_Client_Secret_Here` and replace the existing value with the key you saved during the creation of `webapp` copied from the Azure portal.
+1. Find the key `Enter_Your_WebAPI_Client_ID_Here` and replace the existing value with the application ID (clientId) of `resource-api` app copied from the Azure portal.
 
-## Running the sample
+### Step 4: Running the sample
 
+From your shell or command line, execute the following commands:
+
+1. Open a terminal or the integrated VSCode terminal.
+1. In the root directory as this project, run `cd resource-api`. 
+1. run `mvn clean compile spring-boot:run`.
 1. Open a terminal or the integrated VSCode terminal.
-1. In the same directory as this readme file, run `cd resource-api`. Then run `mvn clean compile spring-boot:run`.
-1. Open a second terminal tab or the another integrated VSCode terminal.
-1. In the same directory as this readme file, run `cd webapp`. Then run `mvn clean compile spring-boot:run`.
+1. In the root directory as this project, run `cd webapp`. 
+1. run `mvn clean compile spring-boot:run`.
 1. Open your browser and navigate to `http://localhost:8080`.
 
```

Comment:
this section needs love from code gen, refer to 4-1 readme.. like in 4-1 it prints as follows



### Step 4: Running the sample

From your shell or command line, execute the following commands:

```console
    cd 4-WebApp-Your-API\4-1-MyOrg\TodoListService
    dotnet run
```

Then, open a separate command terminal and run: 

```console
    cd 4-WebApp-Your-API\4-1-MyOrg\Client
    dotnet run
```